MovieApp/views/MovieDetails/MovieDetailsWithCoverView.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  MovieDetailsWithCoverView.swift
+//  MovieApp
+//
+//  Created by Milan Vrankić on 28.03.2023..
+//
+
+import Foundation
+import PureLayout
+import MovieAppData
+
+class MovieDetailsWithCoverView: UIView {
+    
+    var coverImage: UIImageView!
+    var rating: UILabel!
+    var userScore: UILabel!
+    var movieTitle: UILabel!
+    var star: UIButton!
+    var length: UILabel!
+    var genere: UILabel!
+    var year: UILabel!
+    var date: UILabel!
+    var category: MovieCategoryModel!
+    var movieDetails: MovieDetailsModel!
+    
+    required init?(coder: NSCoder) {
+        fatalError("coder is not defined!")
+    }
+    
+    init() {
+        super.init(frame: CGRect())
+        movieDetails = MovieUseCase().getDetails(id: 111161)
```

Comment:
You should fetch movie details in the ViewController and forward data to custom views. UIView classes should only contain view related logic.